AppSandboxFileAccess/Classes/AppSandboxFileAccess.m

```diff
@@ -160,15 +205,11 @@ - (BOOL)accessFileURL:(NSURL *)fileUrl withBlock:(AppSandboxFileAccessBlock)bloc
 	
 	// if we have no bookmark data, we need to create it, this may be because our bookmark data was stale, or this is the first time being given permission
 	if (persist && !bookmarkData) {
-		[self persistPermissionURL:allowedUrl];
+		bookmarkData = [self persistPermissionURL:allowedURL];
 	}
 	
-	// execute the block with the file access permissions
-	@try {
-		[allowedUrl startAccessingSecurityScopedResource];
-		block();
-	} @finally {
-		[allowedUrl stopAccessingSecurityScopedResource];
+	if (block && allowedURL && bookmarkData) {
+		block(allowedURL, bookmarkData);
```

Comment:
Turns out this little change here where block is not executed if `bookmarkData` is `nil` has highlighted an existing bug in this code that incorrectly handles stale bookmarks. Several lines above stale bookmarks are handled by recreating the bookmark using the existing `allowedURL`, turns out this will sometimes fail with the `bookmarkData` being `nil` and in that case we're supposed to go back and recreate our URL and create a new bookmark off the new URL object. I'm fixing this bug on another branch.

Because this if statement causes block not to execute (fail silently), I think we should remove the if and instead raise an exception before it if `allowedURL` or `bookmarkData` are still nil at this point. What do you think?